### ISSUE TITLE
CRS and nginx not logging sufficient context

### ISSUE THREAD

**Issue by danieloppenheimer** [SpiderLabs/owasp-modsecurity-crs#1299]
## Type of Issue

Bug

## Description

We are using the core rule set 3.0.2 configured for anomaly scoring.  The following is being blocked by modsecurity:

2019/01/24 10:17:52 [warn] 22326#0: *120 [client 172.16.17.54] ModSecurity: Warning. Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `10' ) [file "/etc/nginx/modsec/owasp-modsecurity-crs-3.0.2/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "36"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 10)"] [data ""] [severity "2"] [ver ""] [maturity "0"] [accuracy "0"] [hostname "1.2.3.4"] [uri "/api/auto-engagement/list/b107e2c8-b4e4-470a-b10d-025b11b376cd"] [unique_id "154834307287.337943"] [ref ""], client: 5.6.7.8, server: test.company.com, request: "DELETE /api/auto-engagement/list/b107e2c8-b4e4-470a-b10d-025b11b376cd HTTP/1.1", host: "test.company.com", referrer: "https://test.company.com/studio/story/d5fac22e-e0eb-49ea-8297-a0ec11ce1149" 

But I do not see any other log message in either the nginx error.log or the modsecurity audit log.  As rule 949110 is the rule which determines whether or not the anomaly score is high enough to be blocked, I would expect to see more context in one or both of those files.  That is, I would expect to see one or more log message for the rules that triggered the high anomaly score.  I have set DELETE to an allowed header in the crs-setup.conf file:

SecAction \
 "id:900200,\
  phase:1,\
  nolog,\
  pass,\
  t:none,\
  setvar:'tx.allowed_methods=GET HEAD POST OPTIONS PUT PATCH DELETE'"

I have also enabled audit logging in the modsecurity.conf file:

SecAuditEngine On

SecAuditLogRelevantStatus "^(?:5|4(?!04))"

# Log everything we know about a transaction.
SecAuditLogParts ABIJDEFHKZ

SecAuditLogType Serial
SecAuditLog /var/log/nginx/modsec_audit.log 

Thanks,

Dan

## Your Environment

* CRS version: v3.0.2
* ModSecurity version: 3.0.2
* Web Server and version: nginx 1.14.0
* Operating System and version:  RHEL 7

## Confirmation

[x ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by theMiddleBlue** [comment]
Hi @danieloppenheimer 

I'm not able to reproduce your issue. I've added `DELETE` on 900200 but your request didn't match the 949110 in my setup:

```
# curl -v -XDELETE 'http://localhost/api/auto-engagement/list/b107e2c8-b4e4-470a-b10d-025b11b376cd'
> DELETE /api/auto-engagement/list/b107e2c8-b4e4-470a-b10d-025b11b376cd HTTP/1.1
> Host: localhost
> User-Agent: curl/7.58.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Tue, 12 Feb 2019 10:36:19 GMT
< Content-Type: text/plain
< Transfer-Encoding: chunked
< Connection: keep-alive
...
```


Can you provide the debug log output related to this request? You can turn on debug logging on your modsecurity.conf with something like this:

```
SecDebugLogLevel 9
SecDebugLog /var/log/modsec-debug.log
```

thanks!

**Comment by fgsch** [comment]
Closing due to inactivity. 

Please reopen or create a new ticket and provide the information requested by @theMiddleBlue .
